services/report/languages/go.py

```diff
@@ -80,71 +78,66 @@ def process_bytes_into_files(
     Line format explanation:
         - https://github.com/golang/go/blob/0104a31b8fbcbe52728a08867b26415d282c35d2/src/cmd/cover/profile.go#L56
         - `name.go:line.column,line.column numberOfStatements count`
-
     """
-    _cur_file = None
-    lines = None
-    ignored_files = []
-    file_name_replacement = {}  # {old_name: new_name}
-    files = {}  # {new_name: <lines defaultdict(list)>}
+
+    files: dict[str, dict[int, set]] = {}
+
     for encoded_line in BytesIO(string):
         line = encoded_line.decode(errors="replace").rstrip("\n")
         if not line or line.startswith("mode: "):
             continue
 
-        # prepare data
-        filename, data = line.split(":", 1)
-        if data.endswith("%"):
-            # File outline e.g., "github.com/nfisher/rsqf/rsqf.go:19: calcP 100.0%"
+        split = line.split(":", 1)
+        # File outline e.g., "github.com/nfisher/rsqf/rsqf.go:19: calcP 100.0%"
+        if len(split) < 2 or not split[1] or split[1].endswith("%"):
             continue
 
-        # if we are on the same file name we can pass this
-        if filename in ignored_files:
-            continue
-
-        if _cur_file != filename:
-            _cur_file = filename
-            if filename in file_name_replacement:
-                filename = file_name_replacement[filename]
-            else:
-                fixed = path_fixer(filename)
-                file_name_replacement[filename] = fixed
-                filename = fixed
-                if filename is None:
-                    ignored_files.append(_cur_file)
-                    _cur_file = None
-                    continue
-
-            lines = files.setdefault(filename, defaultdict(set))
-
-        columns, _, hits = data.split(" ", 2)
-        hits = int(hits)
-        line_start, line_end = columns.split(",", 1)
-        line_start, sc = list(map(int, line_start.split(".", 1)))
+        filename = split[0]
         try:
-            line_end, ec = list(map(int, line_end.split(".", 1)))
+            region = parse_coverage(split[1])
         except ValueError:
+            # FIXME: do we actually want to raise an error here?
+            # Why not just skip over invalid lines, as the coverage file likely
+            # contains other valid lines we can use.
```

Comment:
this is a valid question, and I believe we should indeed "silently" ignore these corrupt lines, though more discussion would be needed here.

Reply:
I think a good compromise for now would be to emit logs that lines have been ignored.

Particularly if coverage starts to fluctuate and questions arise as to "why is Codecov saying my coverage has go up / down?" we can at least point to "your report couldn't be fully parsed"

Another thing we maybe should verify is if the lines are indeed corrupted or if the standard changed a little bit over time and we didn't update our processors accordingly

I don't think being totally silent is the best approach, but I think we should be subtle. As you said, "silently" haha, I agree with that.

Reply:
Indeed, it is hard to tell "complete garbage" from "format has changed".
I’m not quite sure whether using logs (either using `logging` or Sentry) would help us here.
We are dealing with "untrusted user data" in the end, so I also don’t want to be DOS-d by fabricated data.

So for in all cases I have seen, and which have generated internal errors so far, it was "complete garbage".

I think changing the status reporting to:
- File format not recognized at all
- File format fully parsed without errors
- File format parsed, with errors

Having metrics for these cases would also be a good thing as well.